configure fails on gaea

**Describe the bug**
Configure fails on gaea with the following message:
configure: error: The HDF5 version used to build netcdf is incompatible with the set FCFLAGS. NetCDF must be built with a HDF5 version other than 1.14.3 to support floating point exception traps.

Even though I don't have the floating point exception traps flag on

**To Reproduce**

```
module load intel-classic/2022.2.1
module load cray-hdf5/1.12.2.3
module load cray-netcdf/4.9.0.3

autoreconf 
mkdir build
cd build
../configure FC=ftn CC=cc FCFLAGS="-O0 -g" LDFLAGS="-O0 -g"
```

**Expected behavior**
Configure should work

**System Environment**
GAEA with the environment above

**Additional context**
I think the test program should have nc_create instead of nc_open:
https://github.com/NOAA-GFDL/FMS/blob/main/configure.ac#L268

--- a/Makefile.am
+++ b/Makefile.am
@@ -125,10 +125,10 @@ check-code-coverage: check
 .PHONY: check-code-coverage
 
 clean-local:
-	-rm -rf .mods coverage-data coverage-report
+	-rm -rf .mods coverage-data coverage-report test.nc
 else
 clean-local:
-	-rm -rf .mods
+	-rm -rf .mods test.nc
 endif
 
 install-data-hook:

--- a/configure.ac
+++ b/configure.ac
@@ -265,7 +265,7 @@ AC_MSG_CHECKING([if HDF5 version causes floating point exceptions with set flags
 AC_RUN_IFELSE([AC_LANG_PROGRAM([], [[
       use netcdf
       integer i, j
-      j = nf90_open("test.nc", NC_WRITE, i)
+      j = nf90_create("test.nc", NC_WRITE, i)
 ]])], [hdf5_fpe_bug=no], [hdf5_fpe_bug=yes])
 AC_MSG_RESULT([$hdf5_fpe_bug])
 if test $hdf5_fpe_bug = yes; then

--- a/test_fms/monin_obukhov/test_monin_obukhov2.sh
+++ b/test_fms/monin_obukhov/test_monin_obukhov2.sh
@@ -35,7 +35,7 @@ for p in r4 r8
 do
   cp ${top_srcdir}/test_fms/monin_obukhov/input.${p}.nml input.nml
   test_expect_success "test monin_obukhov_mod (${p})" "mpirun -n 1 ./test_monin_obukhov_${p}"
-  rm input.nml
+  rm -f input.nml
 done
 
 test_done

